apps/betterangels-backend/shelters/admin.py

```diff
@@ -793,13 +793,35 @@ class ShelterAdmin(ImportExportModelAdmin):
     search_fields = ("name", "organization__name", "description", "subjective_review")
     resource_class = ShelterResource
 
+    def _get_m2m_permissions(self, action: str = "change") -> dict[str, str]:
+        """
+        Generates a dict of permission names by related_name key for all ManyToManyFields on the Shelter model.
+        """
+        permissions_map = {}
+        for field in Shelter._meta.get_fields():
+            if isinstance(field, models.ManyToManyField):
+                related_model = cast(Type[models.Model], field.related_model)
+                model_name = related_model._meta.model_name  # singular name
+                permission_codename = f"{action}_{model_name}"
+                permissions_map[field.name] = f"shelters.{permission_codename}"
+
+        return permissions_map
+
     def get_readonly_fields(
         self, request: HttpRequest, obj: Optional[Shelter] = None
     ) -> Union[list[str], Tuple[str, ...]]:
         readonly_fields = super().get_readonly_fields(request, obj)
         readonly_fields = (*readonly_fields, "updated_at", "updated_by")
         if not request.user.has_perm(ShelterFieldPermissions.CHANGE_IS_REVIEWED):
             readonly_fields = (*readonly_fields, "is_reviewed")
+
+        all_permissions = request.user.get_all_permissions()
+
+        if no_permission_fields := [
+            field_name for field_name, perm in self._get_m2m_permissions().items() if perm not in all_permissions
+        ]:
+            readonly_fields = (*readonly_fields, *no_permission_fields)
```

Comment:
this fixes a bug where users were able to change fields that they didn't have privledges to change... making them readonly instead of blocking the update inside the ShelterForm